Easydict/Feature/Service/Caiyun/CaiyunService.swift

```diff
@@ -0,0 +1,152 @@
+//
+//  CaiyunService.swift
+//  Easydict
+//
+//  Created by Kyle on 2023/11/7.
+//  Copyright © 2023 izual. All rights reserved.
+//
+
+import Alamofire
+import Foundation
+
+@objc(EZCaiyunService)
+public final class CaiyunService: QueryService {
+    override public func serviceType() -> ServiceType {
+        .caiyun
+    }
+
+    override public func link() -> String? {
+        "https://fanyi.caiyunapp.com"
+    }
+
+    override public func name() -> String {
+        NSLocalizedString("caiyun_translate", comment: "The name of Caiyun Translate")
+    }
+
+    override public func supportLanguagesDictionary() -> MMOrderedDictionary<AnyObject, AnyObject> {
+        // TODO: Replace MMOrderedDictionary in the API
+        let orderedDict = MMOrderedDictionary<AnyObject, AnyObject>()
+        orderedDict.setObject("auto" as NSString, forKey: Language.auto.rawValue as NSString)
+        orderedDict.setObject("zh" as NSString, forKey: Language.simplifiedChinese.rawValue as NSString)
+        orderedDict.setObject("en" as NSString, forKey: Language.english.rawValue as NSString)
+        orderedDict.setObject("ja" as NSString, forKey: Language.japanese.rawValue as NSString)
```

Comment:
尝试了一下，好像可以改成这样，这种用 Copilot 自动补全也很舒服 😌
```swift
        let keys = ["auto", "zh", "en", "ja"]
        let values = [Language.auto, Language.simplifiedChinese, Language.english, Language.japanese]
        
        for (index, key) in keys.enumerated() {
            orderedDict.setObject(key as NSString, forKey: values[index].rawValue as NSString)
        }
```

Reply:
strongly -1 for this.

The problem is the legacy code have a dependency on MMOrderedDictionary which does not conforms to `ExpressibleByDictionaryLiteral`

I have 2 suggestions for your snippets.

1️⃣
```swift
let keys = ["auto", "zh", "en", "ja"]
let values: [Language] = [.auto, .simplifiedChinese, .english, .japanese]

for (index, key) in keys.enumerated() {
    orderedDict.setObject(key as NSString, forKey: values[index].rawValue as NSString)
}
```

2️⃣
The values.count is not guaranteed be the same as keys.count and it may crash at runtime.
```swift
let dic: [String: Language] = ["auto": .auto, ...]
dic.forEach { key, value in
    orderedDict.setObject(value as NSString, forKey: key.rawValue as NSString)
}
```

IMO, I'd like to use 2️⃣ or my original code for more flexibility.

Reply:
> 我对 Swift 不太熟，但总感觉这个类型强转有点繁琐，如果支持很多种语言，这里写起来会很累，没道理 Swift 比 OC 还别扭。有没有办法能优化一下？

That's because MMOrderedDictionary has 2 ObjC generic requirements (ObjC generic has a implicit constraint of AnyObject. String is a struct/value type. So an explicit bridge statement `as NSString` is needed in this context.

Reply:
明白了，这里有官方库实现自然是好的，这次提交暂时不改，稍后我弄个 issue 给你替换 MMOrderedDictionary，提交到 swift-migration 分支。